src/core/cheats.c

```diff
@@ -802,3 +803,30 @@ void mCheatDeviceDeinit(struct mCPUComponent* component) {
 		}
 	}
 }
+
+void mCheatToggle(struct mCheatDevice* device) {
+	size_t i;
+	bool cheatsDisabled = true;
+	for (i = 0; i < mCheatSetsSize(&device->cheats); ++i) {
+		struct mCheatSet* cheat = *mCheatSetsGetPointer(&device->cheats, i);
+		if (cheat->enabled) {
+			_unpatchROM(device, cheat);
+			cheat->enabled = false;
+			cheat->savedToggle = true;
+			cheatsDisabled = false;
+		} else if (device->resetSaved) {
```

Comment:
device->resetSaved is to check if the cheats have been modified. If so, then clear any saved cheats.